### PR TITLE
Update README

### LINKED ISSUE #118: Document newly added features

### PR DESCRIPTION
**Pull request by rogermparent**
This PR has a few goals
- Un-document features were removed in #33, which is now merged. This only really includes the PostCSS options, as the layouts system wasn't documented in the first place.
- document Gatsby features added in #33 in README and `pluginOptionsSchema`
- Move README to the actual package's subdirectory, and give the monorepo a mini-README pointing to it (lets README go to npm)

fixes #118, but just barely. We can probably do with more elaboration and formatting.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,14 +24,6 @@ to change that as we make improvements and iron out issues!
...
   passed to the simpleLinker remark plugin
 
-- cssBase: string Used as base files for global PostCSS variables and queries
-
-- customMediaConfig: object config passed to `postcss-custom-media`
-
-- customPropertiesConfig: object config passed to `postcss-custom-properties`
-
-- colorModConfig: object config passed to `postcss-color-mod`
-
```
Comment (yathomasi) [inline README.md:34]: I think a few other options were added too on #33. Can we also add those, or do we want to add them through another PR?

Comment (rogermparent) [reply README.md:34]: I'll open a draft ticket on the board and assign myself for a follow-up after I'm done with plausible

Comment (rogermparent) [reply README.md:34]: Actually, for now I'll add a super-basic list so we at least have them. I'll leave the ticket for more elaboration later.

Comment (rogermparent) [reply README.md:34]: Just added the new commands and some more elaboration on others. Still needs some work, but it'll serve as a base.

File: README.md [hunk 2]
```diff
@@ -9,36 +9,110 @@ to change that as we make improvements and iron out issues!
...
-- customMediaConfig: object config passed to `postcss-custom-media`
+  ```ts
+  const defaultGetTemplate = (template, defaultTemplate) =>
+    template
+      ? require.resolve(path.resolve('src', 'templates', template + '.tsx'))
+      : defaultTemplate
+  ```
 
-- customPropertiesConfig: object config passed to `postcss-custom-properties`
+  This function will be used to
```
Comment (yathomasi) [inline README.md:37]: I believe this is incomplete.

**Review by yathomasi — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (rogermparent) [reply README.md:37]: :facepalm:

File: packages/gatsby-theme-iterative/gatsby-node.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ exports.pluginOptionsSchema = ({ Joi }) => {
     ),
     remark: Joi.boolean().default(true),
     filesystem: Joi.boolean().default(true),
-    glossaryDirectory: Joi.string().default(
+    glossaryPath: Joi.string().default(
       path.resolve('content', 'docs', 'user-guide', 'basic-concepts')
```
Comment (rogermparent) [inline packages/gatsby-theme-iterative/gatsby-node.js:20]: incomplete rename from before

**Pull request merged**